api/src/pcapi/core/providers/titelive_bq_music_search.py

```diff
@@ -0,0 +1,77 @@
+from pcapi.core.providers.titelive_music_search import is_music_codesupport_allowed
+from pcapi.core.providers.titelive_music_search import parse_titelive_music_codesupport
+from pcapi.core.providers.titelive_music_search import parse_titelive_music_genre
+
+
+logger = logging.getLogger(__name__)
+
+
+class BigQueryTiteliveMusicProductSync(BigQuerySyncTemplate[BigQueryTiteliveMusicProductModel]):
+    titelive_base = TiteliveBase.MUSIC
+
+    def get_query(self) -> BaseQuery:
+        return BigQueryTiteliveMusicProductDeltaQuery()
+
+    def filter_batch(
+        self, batch: tuple[BigQueryTiteliveMusicProductModel, ...]
+    ) -> tuple[list[BigQueryTiteliveMusicProductModel], list[str]]:
+        valid_products = []
+        rejected_eans = []
+
+        for bq_product in batch:
+            if not is_music_codesupport_allowed(bq_product.codesupport):
+                if bq_product.ean in self.product_whitelist_eans:
+                    logger.info("Allowing whitelisted music product", extra={"ean": bq_product.ean})
+                else:
+                    rejected_eans.append(bq_product.ean)
+                    logger.info(
+                        "Rejecting ineligible music support",
+                        extra={"ean": bq_product.ean, "support": bq_product.codesupport},
+                    )
+                    continue
+
+            valid_products.append(bq_product)
+
+        return valid_products, rejected_eans
+
+    def fill_product_specifics(
+        self, product: offers_models.Product, bq_product: BigQueryTiteliveMusicProductModel
+    ) -> None:
+        product.subcategoryId = parse_titelive_music_codesupport(bq_product.codesupport).id
+
+        gtl_id = get_gtl_id(bq_product)
+        music_type, music_subtype = parse_titelive_music_genre(gtl_id)
+
+        extra_data = offers_models.OfferExtraData(
+            artist=bq_product.artiste,
+            author=bq_product.compositeur,
+            comment=bq_product.commentaire,
+            contenu_explicite=bq_product.contenu_explicite,
+            date_parution=str(bq_product.dateparution) if bq_product.dateparution else None,
+            dispo=bq_product.dispo,
+            distributeur=bq_product.distributeur,
+            editeur=bq_product.editeur,
+            gtl_id=gtl_id,
+            musicSubType=str(music_subtype.code),
+            musicType=str(music_type.code),
+            music_label=bq_product.label,
+            nb_galettes=bq_product.nb_galettes,
+            performer=bq_product.interprete,
+            prix_musique=str(bq_product.prix) if bq_product.prix is not None else None,
+        )
+
+        clean_extra_data = {key: value for key, value in extra_data.items() if value is not None}
+
+        if product.extraData is None:
+            product.extraData = offers_models.OfferExtraData()
+        product.extraData.update(typing.cast(offers_models.OfferExtraData, clean_extra_data))
```

Comment:
Peut-être que tu peux faire ça ? (ça évite le `typing.cast` et le comprehensive dict)

```suggestion
        product.extraData = product.extraData or offers_models.OfferExtraData()
        for key, value in extra_data.items():
            if value is not None:
                product.extraData[key] = value
```